File: avaloq-exam/src/main/java/com/avaloq/dice/service/DiceRollService.java
package com.avaloq.dice.service;

import com.avaloq.dice.model.DiceResponse;

public interface DiceRollService {
	
	DiceResponse rollDice(Integer numOfDice, Integer numOfSides, Integer totalRolls);
	
	DiceResponse populateTotalSimulations(DiceResponse diceResponse);
	
}

File: avaloq-exam-ui/avaloq-dice/src/app/dice/dice-roll.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Dice } from './dice.model';

@Injectable({
  providedIn: 'root'
})
export class DiceRollService {

  constructor(private http: HttpClient) { }

  rollDice(dice: number, side: number, roll: number){
    return this.http.get<Dice>(`http://localhost:8080/dice?numOfDice=${dice}&numOfSides=${side}&totalRolls=${roll}`);
  };

}
